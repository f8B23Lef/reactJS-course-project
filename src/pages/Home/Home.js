import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import ErrorBoundary from '../../hoc/ErrorBoundary/ErrorBoundary';
import Header from '../../views/Header/Header';
import MainContent from '../../views/MainContent/MainContent';
import Footer from '../../views/Footer/Footer';
import { selectMovieById } from '../../store/moviesSlice';

export default function Home() {
  const [showMovieDetails, setShowMovieDetails] = useState(false);
  const [movieId, setMovieId] = useState();

  const movie = useSelector((state) => selectMovieById(state, movieId));

  const onMovieClick = (movieId) => {
    setShowMovieDetails(true);
    setMovieId(movieId);
    window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
  };

  return (
    <ErrorBoundary>
      <Header
        showMovieDetails={showMovieDetails}
        setShowMovieDetails={setShowMovieDetails}
        movie={movie}
      />
      <MainContent onMovieClick={onMovieClick} />
      <Footer />
    </ErrorBoundary>
  );
}
