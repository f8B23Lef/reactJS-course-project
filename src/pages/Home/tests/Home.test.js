import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Home from '../Home';

jest.mock(
  '../../../hoc/ErrorBoundary/ErrorBoundary',
  () => ({ children }) => children,
);
jest.mock('../../../views/Header/Header', () => () => 'HeaderView');
jest.mock(
  '../../../views/MainContent/MainContent',
  () => () => 'MainContentView',
);
jest.mock('../../../views/Footer/Footer', () => () => 'FooterView');

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => jest.fn(),
}));

describe('Home page', () => {
  it('renders Home page snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );

    expect(asFragment(<Home />)).toMatchSnapshot();
  });
});
