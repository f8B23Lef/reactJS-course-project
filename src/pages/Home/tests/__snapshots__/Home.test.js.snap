// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders Home page snapshot 1`] = `
<DocumentFragment>
  HeaderViewMainContentViewFooterView
</DocumentFragment>
`;
